packages/select-v2/src/useAllowCreate.ts

```diff
@@ -0,0 +1,90 @@
+import { computed, ref } from 'vue'
+import type { ISelectProps } from './token'
+import type { Option } from './select.types'
+
+export function useAllowCreate(props: ISelectProps, states) {
+
+  const createOptionCount = ref(0)
+  const cachedSelectedOption = ref<Option>(null)
+
+  const enableAllowCreateMode = computed(() => {
+    return props.allowCreate && props.filterable
+  })
+
+  function hasExistingOption(query: string) {
+    if (!props.options) {
+      return false
+    }
+    return props.options.concat(states.createdOptions).some(option => {
+      return option.value === query
+    })
+  }
+
+  function selectNewOption(option: Option) {
+    if (!enableAllowCreateMode.value) {
+      return
+    }
+    if (props.multiple && option.created) {
+      states.createdOptions.pop()
+      states.createdOptions.push({
+        ...option,
+        disabled: false,
+      })
```

Comment:
Can you elaborate why popping before pushing?

Reply:
sry, my mistake